### PR TITLE
feat: Payments thank you page

### PR DESCRIPTION
**Pull request by kathleenkhy**
## Problem
Currently the design for the Thank You page for Payments could be better represented for users using these forms for payments, where the look and feel looks too similar to normal forms, breakdown of purchases can be displayed more clearly.

Closes FRM-1558, Closes FRM-1559, Closes FRM-1570, Closes FRM-1569

## Solution
To update the Thank You page for Payment forms to make the design more suited for Payments. To also update feedback section title to be more applicable for payments: 

image

**Breaking Changes** 

- [ ] Yes - this PR contains breaking changes
    - Details ...
- [x] No - this PR is backwards compatible  

**Features**:

Figma link of design: https://www.figma.com/file/CMq0CJtawn7NGM7Zdpi8Ti/Form-Design-Master?type=design&node-id=21679-128953&mode=dev

**BEFORE**:

image

image

**AFTER**:
image

image

## Deploy Notes

**New environment variables**:

- `env var` : env var details

**New scripts**:

- `script` : script details

**New dependencies**:

- `dependency` : dependency details

**New dev dependencies**:

- `dependency` : dependency details

### CODE REVIEW

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/StripeReceiptContainer.tsx [hunk 1]
```diff
@@ -87,11 +87,15 @@ export const StripeReceiptContainer = ({
           paymentDate={paymentReceiptStatus.paymentDate}
         />
       </PaymentStack>
-      <PaymentStack>
-        {!isFeedbackSubmitted && (
-          <FeedbackBlock onSubmit={handleSubmitFeedback} />
-        )}
-      </PaymentStack>
+      {
```
Comment (KenLSM) [inline frontend/src/features/public-form/components/FormPaymentPage/stripe/StripeReceiptContainer.tsx]: Seems to have an extra set of `{}` braces here, is it intended?

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx [hunk 2]
```diff
@@ -34,22 +39,59 @@ const PaymentSummaryRow = ({
...
+  input,
+}: {
+  title: string
+  input: string
+}): JSX.Element => {
+  return (
+    <Stack
+      direction={{ base: 'row', md: 'row' }}
+      spacing={{ base: 0, md: '3rem' }}
+      justify={{ base: 'space-between', md: 'space-between' }}
```
Comment (KenLSM) [inline frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: I see that `base` and `md` are using the same `row` / `space-between`, if we replace them with `direction="row"`, and `justify="space-between"`, would it work?

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx [hunk 3]
```diff
@@ -82,36 +123,64 @@ export const DownloadReceiptBlock = ({
...
+              Summary
+            </Text>
+            <Stack spacing="0.75rem" my="1rem">
+              {products.map((product) => (
+                <LineItem productItem={product} />
+              ))}
+            </Stack>
+            <Divider />
+            <Stack my="1rem">
+              <PaymentSummaryRow title="Total" input={totalAmount} />
```
Comment (KenLSM) [inline frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx:163]: Good practice to create a new component 👍

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx [hunk 4]
```diff
@@ -64,16 +106,15 @@ export const DownloadReceiptBlock = ({
 }: DownloadReceiptBlockProps) => {
   const toast = useToast({ status: 'success', isClosable: true })
 
-  const formattedAmount = useMemo(() => `S$${centsToDollars(amount)}`, [amount])
+  // const formattedAmount = useMemo(() => `S$${centsToDollars(amount)}`, [amount])
```
Comment (KenLSM) [inline frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: Forgot to remove this commented code? 😆

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx [hunk 5]
```diff
@@ -82,36 +123,64 @@ export const DownloadReceiptBlock = ({
...
         <Stack tabIndex={-1} spacing="0.75rem">
-          <Text textStyle="h2" color="content.strong">
-            Your payment has been made successfully.
+          <Text textStyle="h2" color="secondary.500">
+            Thank you, your payment has been made successfully.
           </Text>
-          <Text textStyle="subhead-1" color="content.strong">
-            Your form has been submitted and payment has been made.
+          <Text textStyle="subhead-1" color="secondary.500">
+            This is a message that the admins can customise on their thank you
```
Comment (KenLSM) [inline frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: Is this copy final? So many uses of `This is a message that ...` 🤣

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx [hunk 6]
```diff
@@ -82,36 +123,64 @@ export const DownloadReceiptBlock = ({
     window.location.href = getPaymentInvoiceDownloadUrl(formId, paymentId)
   }
   return (
-    <Box>
-      <Stack spacing="2rem">
+    <>
+      <Box bgColor="white" p="2rem">
```
Comment (KenLSM) [inline frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: /nit can replace `bgColor` with `bg` to save on 5 characters 😆

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx [hunk 7]
```diff
@@ -82,36 +123,64 @@ export const DownloadReceiptBlock = ({
...
+            </Text>
+            <PaymentDetailsRow title="Payment date" input={paymentTimestamp} />
+            <PaymentDetailsRow title="Response ID" input={submissionId} />
+          </Box>
+          <Box
+            bgColor="primary.100"
+            flexDir="row"
+            justifyContent="space-between"
+            mb="1rem"
+            py="1rem" // h="4rem"
```
Comment (KenLSM) [inline frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: Stray `// h="4rem"`?

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx [hunk 8]
```diff
@@ -82,36 +123,64 @@ export const DownloadReceiptBlock = ({
...
-        width={{ base: '100%', md: 'auto' }}
-        leftIcon={<BiDownload fontSize="1.5rem" />}
-        onClick={handleInvoiceClick}
-      >
-        Save proof of payment
-      </Button>
-    </Box>
+        <Stack mx={{ base: '0', md: '1.5rem' }} mt="1.5rem">
+          <Button
+            width={{ base: '100%', md: '100%' }}
```
Comment (KenLSM) [inline frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: Similar to above, `base` and `md` are the same, would it work if we use `w="100%"`?

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/StripeReceiptContainer.tsx (hunk 1 above)
Comment (kathleenkhy) [reply frontend/src/features/public-form/components/FormPaymentPage/stripe/StripeReceiptContainer.tsx]: ahh... I think I was still playing around then when I added it but will remove it!

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx (hunk 2 above)
Comment (kathleenkhy) [reply frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: yes it works!! 🙏🏼

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx (hunk 4 above)
Comment (kathleenkhy) [reply frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: will remove! thanks for catching that :)

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx (hunk 7 above)
Comment (kathleenkhy) [reply frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: will remove 🙏🏼

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx (hunk 8 above)
Comment (kathleenkhy) [reply frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: yes it works! 🙏🏼

File: frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx (hunk 5 above)
Comment (kathleenkhy) [reply frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx]: ahh I think this was an oversight from me - this should be editable by the form admin and should be tied to the issue of allowing the payment thank you page page to be editable (FRM-1560).  I'll put a placeholder text `"Your form has been submitted and payment has been made."` to be hardcoded for now, and will look at how to retrieve the text when working on FRM-1560. 

thank you for catching this too 🙏🏼

File: frontend/src/features/public-form/components/FormEndPage/components/FeedbackBlock.tsx [hunk 9]
```diff
@@ -34,6 +36,12 @@ export const FeedbackBlock = ({
 
   const handleFormSubmit = handleSubmit((inputs) => onSubmit(inputs))
 
+  const { formId, isPaymentEnabled } = usePublicFormContext()
```
Comment (KenLSM) [inline frontend/src/features/public-form/components/FormEndPage/components/FeedbackBlock.tsx]: To remove `formId`

**Review by KenLSM — CHANGES_REQUESTED** [review]
LGTM

**Review by KenLSM — APPROVED** [review]

**Pull request merged**